src/main/java/httpserver/HttpRequest.java

```diff
@@ -0,0 +1,16 @@
+package httpserver;
+
+
+import java.util.*;
+
+public class HttpRequest implements HttpIncomingMessage{
+    private HashMap<String, String> attributes;
+
+    public HttpRequest(HashMap clientRequestAttributes){
+        this.attributes = clientRequestAttributes;
+    }
+
+    public String getRoute(){
+        return attributes.get("route");
```

Comment:
What happens when the `HashMap` doesn't have a `route` key? Should there be a test for that?

Reply:
I'm thinking about making it similar to request, where it's more like a struct that holds values rather than using a`HashMap`.